dataprep/eda/create_report/formatter.py

```diff
@@ -27,6 +27,20 @@
 def format_report(df: pd.DataFrame, mode: Optional[str]) -> Dict[str, Any]:
     """
     Format the data and figures needed by report
+
+    Parameters
+    ----------
+    df
+        The DataFrame for which data are calculated.
+    mode: Optional[str]
+        This controls what type of report to be generated.
+        Currently only the 'basic' is fully implemented.
+
+    Returns
+    -------
+    Dict[str, Any]
+        A dictionary in which formatted data will be stored.
+        This variable acts like an API in passing data to the template engine.
     """
 
     comps: Dict[str, Any] = {}
```

Comment:
Why is an empty dict passed to each `format_` method?

Reply:
`format_` functions will add raw data to this dictionary. Those functions act like a pipeline to the input and return expected output.